# Open MrOS Vitamin D dataset and create combined_df dataframe using Pandas package
OH4Aug17_df is the vitamin D file from QIITA 

In [1]:
from os import listdir
listdir()

['MrOS_Vitamin D_Stat',
 'combined_df.tsv',
 'alphadiv_shannon_visualization.qzv',
 'alphadiv_chao1.tsv',
 'shannon_vector.qza',
 'beta_significance.qzv',
 'feature_table.qza',
 '.DS_Store',
 'metadata.txt',
 'feature_table_export',
 'all.seqs.fa',
 'OLD_alpha_diversity_df.tsv',
 'alphadiv_rarefaction.qzv',
 'braycurtis_filtered.qza',
 'OH4AUG17_CONTENTS copy.PDF',
 'data_feature-table.biom',
 'chao1_OHVD3_viz.png',
 'Supplementary_Code_Verisha_Bhatti_Procedure_File.ipynb',
 'OH4Aug17.csv',
 'combined_df',
 'knightlab-analyses-vitamin-d-66ab000',
 'feature_table.tsv',
 '.ERR2518451.fastq.swp',
 'observed_features_vector.qza',
 'filtered_df.tsv',
 'vitD_seq_data.tsv',
 'alphadiv_shannon.tsv',
 'alphadiv_chao1_visualization.qzv',
 'NEW combined_df.ipynb',
 'visualization_pcoa_jaccard.qzv',
 'Viz',
 'filereport_read_run_ERP107984.tsv',
 'MrOSAUG24',
 'Supplementary_Code_Verisha_Bhatti_Create_Combined_Dataframe.ipynb',
 'chao1_OHVD3.svg',
 'braycurtis_results.qza',
 'metadata.csv',
 '.ipyn

In [2]:
import pandas as pd 
data = pd.read_sas('./MrOS_Vitamin D_Stat/OH4AUG17/OH4Aug17.sas7bdat')
data.to_csv("OH4Aug17.csv")

In [3]:
import pandas as pd 
OH4Aug17_df = pd.read_csv('OH4Aug17.csv',sep = ",")
type(OH4Aug17_df)

pandas.core.frame.DataFrame

In [4]:
OH4Aug17_df.head()

,Unnamed: 0,ID,OHV1D3,OHV24D3,OHVD3,OHVD2,SITE,OHV1D2,OHV1D2CT,OHVD2CT,OHVDTOT,OHV1DTOT,OHVD2I,OHVD3I,OHV1D2I,OHV1D3I,OHV24D3I,OHMONTH,OHSEAS
0,0,b'BI0023',39.3,1.77,25.8,0.0,b'BI',0.0,1.0,1.0,25.8,39.3,0.0,64.3968,0.0,98.0928,4.41792,9.0,3.0
1,1,b'BI0056',61.9,3.91,39.2,0.0,b'BI',0.0,1.0,1.0,39.2,61.9,0.0,97.8432,0.0,154.5024,9.75936,5.0,2.0
2,2,b'BI0131',52.1,1.49,23.1,0.0,b'BI',0.0,1.0,1.0,23.1,52.1,0.0,57.6576,0.0,130.0416,3.71904,4.0,2.0
3,3,b'BI0153',43.1,2.14,27.3,0.0,b'BI',0.0,1.0,1.0,27.3,43.1,0.0,68.1408,0.0,107.5776,5.34144,4.0,2.0
4,4,b'BI0215',50.2,3.62,33.0,0.0,b'BI',0.0,1.0,1.0,33.0,50.2,0.0,82.3680,0.0,125.2992,9.03552,10.0,4.0


### Data Cleaning

In [5]:
#remove "Unnamed: 0" column 
OH4Aug17_df = OH4Aug17_df.drop(columns=['Unnamed: 0'])

In [6]:
#remove b' in front of ID column
OH4Aug17_df["ID"] = OH4Aug17_df["ID"].str.replace("b'", "").str.replace("'", "")

#set ID column as index
OH4Aug17_df.set_index("ID")

,OHV1D3,OHV24D3,OHVD3,OHVD2,SITE,OHV1D2,OHV1D2CT,OHVD2CT,OHVDTOT,OHV1DTOT,OHVD2I,OHVD3I,OHV1D2I,OHV1D3I,OHV24D3I,OHMONTH,OHSEAS
ID,,,,,,,,,,,,,,,,,
BI0023,39.3,1.77,25.8,0.0,b'BI',0.0,1.0,1.0,25.8,39.3,0.0,64.3968,0.0,98.0928,4.41792,9.0,3.0
BI0056,61.9,3.91,39.2,0.0,b'BI',0.0,1.0,1.0,39.2,61.9,0.0,97.8432,0.0,154.5024,9.75936,5.0,2.0
BI0131,52.1,1.49,23.1,0.0,b'BI',0.0,1.0,1.0,23.1,52.1,0.0,57.6576,0.0,130.0416,3.71904,4.0,2.0
BI0153,43.1,2.14,27.3,0.0,b'BI',0.0,1.0,1.0,27.3,43.1,0.0,68.1408,0.0,107.5776,5.34144,4.0,2.0
BI0215,50.2,3.62,33.0,0.0,b'BI',0.0,1.0,1.0,33.0,50.2,0.0,82.3680,0.0,125.2992,9.03552,10.0,4.0
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
SD8996,41.8,1.58,20.8,0.0,b'SD',0.0,1.0,1.0,20.8,41.8,0.0,51.9168,0.0,104.3328,3.94368,11.0,4.0
SD8999,55.1,3.91,36.2,0.0,b'SD',0.0,1.0,1.0,36.2,55.1,0.0,90.3552,0.0,137.5296,9.75936,12.0,4.0
SD9001,42.6,3.16,37.6,0.0,b'SD',0.0,1.0,1.0,37.6,42.6,0.0,93.8496,0.0,106.3296,7.88736,4.0,2.0


In [7]:
#Meta data is our mapping file from QIITA
metadata_df = pd.read_csv('metadata.csv',sep = ",")
metadata_df.set_index("anonymized_name")
metadata_df

,sample_name,anonymized_name,collection_timestamp,description,dna_extracted,elevation,empo_1,empo_2,empo_3,empo_4,...,latitude,life_stage,longitude,physical_specimen_location,physical_specimen_remaining,qiita_study_id,scientific_name,sex,taxon_id,title
0,11274.BI0023,BI0023,restricted access,stool sample BI0023,True,196,Host-associated,Host-associated (non-saline),Animal (non-saline),Animal distal gut (non-saline),...,33.520661,adult,-86.80249,OHSU,False,11274,human gut metagenome,male,408170,MrOS vitamin D
1,11274.BI0056,BI0056,restricted access,stool sample BI0023,True,196,Host-associated,Host-associated (non-saline),Animal (non-saline),Animal distal gut (non-saline),...,33.520661,adult,-86.80249,OHSU,False,11274,human gut metagenome,male,408170,MrOS vitamin D
2,11274.BI0131,BI0131,restricted access,stool sample BI0023,True,196,Host-associated,Host-associated (non-saline),Animal (non-saline),Animal distal gut (non-saline),...,33.520661,adult,-86.80249,OHSU,False,11274,human gut metagenome,male,408170,MrOS vitamin D
3,11274.BI0152,BI0152,restricted access,stool sample BI0023,True,196,Host-associated,Host-associated (non-saline),Animal (non-saline),Animal distal gut (non-saline),...,33.520661,adult,-86.80249,OHSU,False,11274,human gut metagenome,male,408170,MrOS vitamin D
4,11274.BI0153,BI0153,restricted access,stool sample BI0023,True,196,Host-associated,Host-associated (non-saline),Animal (non-saline),Animal distal gut (non-saline),...,33.520661,adult,-86.80249,OHSU,False,11274,human gut metagenome,male,408170,MrOS vitamin D
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
917,11274.SD9003,SD9003,restricted access,stool sample SD9009,True,193,Host-associated,Host-associated (non-saline),Animal (non-saline),Animal distal gut (non-saline),...,32.715738,adult,-117.16110,OHSU,False,11274,human gut metagenome,male,408170,MrOS vitamin D
918,11274.SD9008,SD9008,restricted access,stool sample SD9009,True,193,Host-associated,Host-associated (non-saline),Animal (non-saline),Animal distal gut (non-saline),...,32.715738,adult,-117.16110,OHSU,False,11274,human gut metagenome,male,408170,MrOS vitamin D
919,11274.SD9009,SD9009,restricted access,stool sample SD9009,True,193,Host-associated,Host-associated (non-saline),Animal (non-saline),Animal distal gut (non-saline),...,32.715738,adult,-117.16110,OHSU,False,11274,human gut metagenome,male,408170,MrOS vitamin D
920,11274.SD9011,SD9011,restricted access,stool sample SD9009,True,193,Host-associated,Host-associated (non-saline),Animal (non-saline),Animal distal gut (non-saline),...,32.715738,adult,-117.16110,OHSU,False,11274,human gut metagenome,male,408170,MrOS vitamin D


In [31]:
combined_df = metadata_df.join(OH4Aug17_df, how = 'inner')
combined_df

,sample_name,anonymized_name,collection_timestamp,description,dna_extracted,elevation,empo_1,empo_2,empo_3,empo_4,...,OHVD2CT,OHVDTOT,OHV1DTOT,OHVD2I,OHVD3I,OHV1D2I,OHV1D3I,OHV24D3I,OHMONTH,OHSEAS
0,11274.BI0023,BI0023,restricted access,stool sample BI0023,True,196,Host-associated,Host-associated (non-saline),Animal (non-saline),Animal distal gut (non-saline),...,1.0,25.8,39.3,0.0,64.3968,0.0,98.0928,4.41792,9.0,3.0
1,11274.BI0056,BI0056,restricted access,stool sample BI0023,True,196,Host-associated,Host-associated (non-saline),Animal (non-saline),Animal distal gut (non-saline),...,1.0,39.2,61.9,0.0,97.8432,0.0,154.5024,9.75936,5.0,2.0
2,11274.BI0131,BI0131,restricted access,stool sample BI0023,True,196,Host-associated,Host-associated (non-saline),Animal (non-saline),Animal distal gut (non-saline),...,1.0,23.1,52.1,0.0,57.6576,0.0,130.0416,3.71904,4.0,2.0
3,11274.BI0152,BI0152,restricted access,stool sample BI0023,True,196,Host-associated,Host-associated (non-saline),Animal (non-saline),Animal distal gut (non-saline),...,1.0,27.3,43.1,0.0,68.1408,0.0,107.5776,5.34144,4.0,2.0
4,11274.BI0153,BI0153,restricted access,stool sample BI0023,True,196,Host-associated,Host-associated (non-saline),Animal (non-saline),Animal distal gut (non-saline),...,1.0,33.0,50.2,0.0,82.3680,0.0,125.2992,9.03552,10.0,4.0
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
562,11274.PO6848,PO6848,restricted access,stool sample PO6836,True,15,Host-associated,Host-associated (non-saline),Animal (non-saline),Animal distal gut (non-saline),...,1.0,20.8,41.8,0.0,51.9168,0.0,104.3328,3.94368,11.0,4.0
563,11274.PO6858,PO6858,restricted access,stool sample PO6836,True,15,Host-associated,Host-associated (non-saline),Animal (non-saline),Animal distal gut (non-saline),...,1.0,36.2,55.1,0.0,90.3552,0.0,137.5296,9.75936,12.0,4.0
564,11274.PO6862,PO6862,restricted access,stool sample PO6836,True,15,Host-associated,Host-associated (non-saline),Animal (non-saline),Animal distal gut (non-saline),...,1.0,37.6,42.6,0.0,93.8496,0.0,106.3296,7.88736,4.0,2.0
565,11274.PO6863,PO6863,restricted access,stool sample PO6836,True,15,Host-associated,Host-associated (non-saline),Animal (non-saline),Animal distal gut (non-saline),...,1.0,32.0,81.3,0.0,79.8720,0.0,202.9248,8.46144,11.0,4.0


In [32]:
#check list of column names in new dataframe: combined_df
list(combined_df.columns)

['sample_name',
 'anonymized_name',
 'collection_timestamp',
 'description',
 'dna_extracted',
 'elevation',
 'empo_1',
 'empo_2',
 'empo_3',
 'empo_4',
 'env_biome',
 'env_feature',
 'env_material',
 'env_package',
 'geo_loc_name',
 'host_age',
 'host_age_units',
 'host_body_habitat',
 'host_body_mass_index',
 'host_body_product',
 'host_body_site',
 'host_common_name',
 'host_height',
 'host_height_units',
 'host_scientific_name',
 'host_subject_id',
 'host_taxid',
 'host_weight',
 'host_weight_units',
 'latitude',
 'life_stage',
 'longitude',
 'physical_specimen_location',
 'physical_specimen_remaining',
 'qiita_study_id',
 'scientific_name',
 'sex',
 'taxon_id',
 'title',
 'ID',
 'OHV1D3',
 'OHV24D3',
 'OHVD3',
 'OHVD2',
 'SITE',
 'OHV1D2',
 'OHV1D2CT',
 'OHVD2CT',
 'OHVDTOT',
 'OHV1DTOT',
 'OHVD2I',
 'OHVD3I',
 'OHV1D2I',
 'OHV1D3I',
 'OHV24D3I',
 'OHMONTH',
 'OHSEAS']

### Note on QIIME2 IDs
- QIIME2 uses IDs with the suffix "11274." added before the actual ID, thus for the combined_df I will change the index to be the sample_name column which has IDs with this suffix added, rather than the actual ID
- QIIME2 also requires that the 1st column be named "sampleid" and I will rename the column

In [36]:
#QIIME2 requires the ID column to be renamed to sampleid
combined_df = combined_df.rename(columns={'sample_name': 'sampleid'})

In [37]:
#set "sampleid as index"
combined_df.set_index("sampleid")

,anonymized_name,collection_timestamp,description,dna_extracted,elevation,empo_1,empo_2,empo_3,empo_4,env_biome,...,OHVD2CT,OHVDTOT,OHV1DTOT,OHVD2I,OHVD3I,OHV1D2I,OHV1D3I,OHV24D3I,OHMONTH,OHSEAS
sampleid,,,,,,,,,,,,,,,,,,,,,
11274.BI0023,BI0023,restricted access,stool sample BI0023,True,196,Host-associated,Host-associated (non-saline),Animal (non-saline),Animal distal gut (non-saline),urban biome,...,1.0,25.8,39.3,0.0,64.3968,0.0,98.0928,4.41792,9.0,3.0
11274.BI0056,BI0056,restricted access,stool sample BI0023,True,196,Host-associated,Host-associated (non-saline),Animal (non-saline),Animal distal gut (non-saline),urban biome,...,1.0,39.2,61.9,0.0,97.8432,0.0,154.5024,9.75936,5.0,2.0
11274.BI0131,BI0131,restricted access,stool sample BI0023,True,196,Host-associated,Host-associated (non-saline),Animal (non-saline),Animal distal gut (non-saline),urban biome,...,1.0,23.1,52.1,0.0,57.6576,0.0,130.0416,3.71904,4.0,2.0
11274.BI0152,BI0152,restricted access,stool sample BI0023,True,196,Host-associated,Host-associated (non-saline),Animal (non-saline),Animal distal gut (non-saline),urban biome,...,1.0,27.3,43.1,0.0,68.1408,0.0,107.5776,5.34144,4.0,2.0
11274.BI0153,BI0153,restricted access,stool sample BI0023,True,196,Host-associated,Host-associated (non-saline),Animal (non-saline),Animal distal gut (non-saline),urban biome,...,1.0,33.0,50.2,0.0,82.3680,0.0,125.2992,9.03552,10.0,4.0
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
11274.PO6848,PO6848,restricted access,stool sample PO6836,True,15,Host-associated,Host-associated (non-saline),Animal (non-saline),Animal distal gut (non-saline),urban biome,...,1.0,20.8,41.8,0.0,51.9168,0.0,104.3328,3.94368,11.0,4.0
11274.PO6858,PO6858,restricted access,stool sample PO6836,True,15,Host-associated,Host-associated (non-saline),Animal (non-saline),Animal distal gut (non-saline),urban biome,...,1.0,36.2,55.1,0.0,90.3552,0.0,137.5296,9.75936,12.0,4.0
11274.PO6862,PO6862,restricted access,stool sample PO6836,True,15,Host-associated,Host-associated (non-saline),Animal (non-saline),Animal distal gut (non-saline),urban biome,...,1.0,37.6,42.6,0.0,93.8496,0.0,106.3296,7.88736,4.0,2.0


In [38]:
# save combined_df as a tsv file for QIIME2
import pandas as pd

combined_df.to_csv('combined_df.tsv',sep = "\t") 